packages/frontend/src/components/dialogs/ProxyConfiguration/index.tsx

```diff
@@ -0,0 +1,403 @@
+  }, [accountId, configured])
+
+  /**
+   * Update proxy settings in the backend
+   * called after any proxyState change
+   */
+  useEffect(() => {
+    if (!proxyState.updateSettings) {
+      // don't update while loading values from backend
+      return
+    }
+    const updateProxySettings = async () => {
+      const proxyString =
+        proxyState.proxies.length > 0
+          ? [
+              proxyState.activeProxy,
+              ...proxyState.proxies.filter(p => p !== proxyState.activeProxy),
+            ]
+              .filter(s => !!s) // remove null & empty strings
+              .join('\n')
+          : ''
+      if (proxyState.enabled && proxyString.trim() === '') {
+        openAlertDialog({
+          message: tx('proxy_invalid'),
+        })
+        return
+      }
+      try {
+        await BackendRemote.rpc.batchSetConfig(accountId, {
+          proxy_url: proxyString,
+          proxy_enabled: proxyState.enabled ? Proxy.ENABLED : Proxy.DISABLED,
+        })
+
+        await BackendRemote.rpc.stopIo(accountId)
+        await BackendRemote.rpc.startIo(accountId)
+      } catch (error) {
+        log.error('failed to update proxy settings', error)
+        openAlertDialog({
+          message: unknownErrorToString(error),
+        })
+      }
+    }
+    updateProxySettings()
+  }, [
+    proxyState.enabled,
+    accountId,
+    openAlertDialog,
+    tx,
+    proxyState.proxies,
+    proxyState.activeProxy,
+    proxyState.updateSettings,
+    proxyState,
+  ])
+
+  /**
+   * validate settings before closing
+   * to avoid invalid settings
+   */
+  const closeDialog = () => {
+    if (proxyState.enabled && proxyState.proxies.length === 0) {
+      openAlertDialog({
+        message: tx('proxy_invalid'),
+      })
+      return
+    }
+    onClose()
+  }
+
+  const copyToClipboard = useCallback(
+    (url: string) => {
+      navigator.clipboard.writeText(url)
+      openAlertDialog({
+        message: `${url}\n${tx('copied_to_clipboard')}`,
+      })
+    },
+    [tx, openAlertDialog]
+  )
+
+  // some basic validations, returns true if the url seems valid
+  // and is not already in the list of proxies
+  const maybeValidProxyUrl = (url: string): boolean => {
+    const parts = url.split('://')
+    return (
+      parts.length === 2 &&
+      parts[0].length >= 2 && // shortest protocol is ss://
+      parts[1].length >= 1 && // host
+      !proxyState.proxies.includes(url)
+    )
+  }
+
+  return (
+    <Dialog
+      fixed
+      width={400}
+      dataTestid='proxy-dialog'
+      canOutsideClickClose={false}
+    >
+      <DialogHeader
+        title={tx('menu_settings')}
+        onClose={closeDialog}
+        dataTestid='proxy-settings'
+      />
+      <DialogBody className={styles.proxyDialogBody}>
+        <div className={styles.container}>
+          <SettingsSwitch
+            label={tx('proxy_use_proxy')}
+            value={proxyState.enabled}
+            onChange={changeProxyEnable}
+          />
```

Comment:
When you open the dialog for the first time, you have no proxies, and when you toggle this on, you get an error message right away. Not a nice experience.
FYI Android opens the "add proxy" dialog in such a case.

Reply:
Well you could also argue that you can't enable proxy if you have not yet entered a proxy URL. We could hide the enable button and only show it after you added a proxy URL.

Reply:
Well sure, logically you can't, but the UX is not the best IMO. (I don't think this issue is a reason not to merge the MR BTW).